src/components/Navigator/NavigatorDataProvider.vue

```diff
@@ -73,8 +86,7 @@ export default {
     async fetchIndexData() {
       try {
         this.isFetching = true;
-        const { interfaceLanguages } = await fetchIndexPathsData();
-        this.navigationIndex = Object.freeze(interfaceLanguages);
```

Comment:
I would still freeze the entire `navigationIndex`, because it can be enormous, yet we never intend to selectively update it, as I mentioned before, so there is no need for deep reactivity tracking.